crt/enter.c

```diff
@@ -45,6 +49,14 @@ long myst_syscall(long n, long params[6])
     if (n == SYS_setitimer)
         pthread_once(&_once, _create_itimer_thread);
 
+    if (n == SYS_fork)
```

Comment:
actually we can implement it in the kernel, but we can also implement it in application space also.

Reply:
implementing in the kernel is now a little more complicated due to the switching stack

Reply:
regarding the locks, well there used to be something in there, but apparently not any more so the lock is gone.

Reply:
We implemented fork in the kernel because there is no way to create a thread inside the kernel today and implementing it would be very complex. So we implement fork inside the CRT by calling ``clone``, which enters the kernel. This is how ``posix_spawn`` is implemented in musl libc (it calls clone and all synchronization occurs in the CRT).

Reply:
We can make ocall to create threads. I think we just need to copy some code  in `pthread_create` to ensure the thread data conforms to what MUSL expects.

Reply:
The must libc pthread structures are opaque to the kernel. We will need some design discussions later.